core/tool_cache.py

```
@@ -0,0 +1,108 @@
+    """
+    LRU 缓存：存储工具完整结果，用摘要替代原文传给 LLM。
+
+    原则（per Claude cookbook）：保留"调用发生过"的记录，丢弃原文。
+    """
+
+    def __init__(self, max_size: int = 50):
+        self._cache: OrderedDict[str, Dict[str, Any]] = OrderedDict()
+        self._max_size = max_size
+
+    def store(self, result: Dict[str, Any]) -> str:
+        """存入缓存，返回 cache_key。"""
```

Ruff (0.15.10)

[warning] Docstring contains ambiguous `，` (FULLWIDTH COMMA). Did you mean `,` (COMMA)?

(RUF002)